cmake/jsoncpp.cmake

```diff
@@ -10,8 +10,13 @@ set(prefix "${CMAKE_BINARY_DIR}/deps")
 set(JSONCPP_LIBRARY "${prefix}/lib/${CMAKE_STATIC_LIBRARY_PREFIX}jsoncpp${CMAKE_STATIC_LIBRARY_SUFFIX}")
 set(JSONCPP_INCLUDE_DIR "${prefix}/include")
 
-if(NOT MSVC)
-    set(JSONCPP_EXTRA_FLAGS "-std=c++11")
+# TODO: Investigate why this breaks some emscripten builds and
+# check whether this can be removed after updating the emscripten
+# versions used in the CI runs.
+if(EMSCRIPTEN)
```

Comment:
Previously `c++11` was added for all compilers bar MSVC, now it is not only added for EMSCRIPTEN.

Is that the intention?

Reply:
Yes, that's intended. There's no actual change in when ``-std=c++11`` is used:
  - EMSCRIPTEN is not MSVC, so it was added for emscripten
  - for NOT EMSCRIPTEN, CMAKE_CXX_FLAGS contain it for all compilers but MSVC

So it should still always be there except for MSVC.

Reply:
Well, to be fair, it is different for "unknown compilers", but there's no reason to think that an unknown compiler *does* require ``-std=c++11`` here...

Reply:
In that case why to have this extra condition for emscripten here? It falls under gcc/clang, iirc?

Reply:
Ah ok, I guess I've missed the part this removes all other flags. Can you leave a comment explaining it (e.g "drop all other flags")?

Reply:
Most of those flags (starting with `-s`) are I think valid for the linking stage.

Reply:
Yes, that's why I would guess that it's not the ``-s`` flags. Currently I would guess ``-fvisibility=hidden``, but I haven't looked into it, yet. In any case I'll add another explanatory comment here.

Reply:
Added another comment below.

Reply:
Well, I wouldn't say they are "silently dropped", since none of the flags were added to jsoncpp at all before :). But sure, we should fix the issue - for that we should look into which flags are responsible and why it depends on the emscripten version, though. That's why my idea was to merge this with the workaround now and to look into the emscripten case in more detail later on. But if you prefer, I can look into it right away.

Reply:
By your previous sentence ("this is what caused build failures for the travis emscripten build earlier") I assumed they were indeed passed through.

Reply:
They were in the original version of this PR (that's why I changed it). But before this PR, they never were.

Reply:
Thanks for the explanation :)